File: flutter/core/lib/src/network/network_service.dart
/// Network service contract.
abstract class NetworkService {
  Future<Map<String, dynamic>> get(String endpoint);
  Future<Map<String, dynamic>> post(String endpoint, Map<String, dynamic> body);
}

/// Stub implementation for scaffolding.
class StubNetworkService implements NetworkService {
  @override
  Future<Map<String, dynamic>> get(String endpoint) async => {};

  @override
  Future<Map<String, dynamic>> post(
          String endpoint, Map<String, dynamic> body) async =>
      {};
}

File: ios/Packages/Core/Sources/Core/Network/NetworkService.swift
import Foundation

/// Network service protocol.
public protocol NetworkService: AnyObject {
    func get(endpoint: String) async throws -> [String: Any]
    func post(endpoint: String, body: [String: Any]) async throws -> [String: Any]
}

/// Stub implementation for scaffolding.
public final class StubNetworkService: NetworkService {
    public init() {}

    public func get(endpoint: String) async throws -> [String: Any] { [:] }
    public func post(endpoint: String, body: [String: Any]) async throws -> [String: Any] { [:] }
}
